dali/test/python/test_operator_coord_flip.py

```diff
@@ -0,0 +1,90 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali as dali
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+from functools import partial
+from test_utils import check_batch
+from test_utils import compare_pipelines
+from test_utils import RandomDataIterator
+import math
+from nose.tools import *
+
+class CoordFlipPipeline(Pipeline):
+    def __init__(self, device, batch_size, iterator, layout,
+                 num_threads=1, device_id=0):
+        super(CoordFlipPipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+        self.iterator = iterator
+        self.coord_flip = ops.CoordFlip(device = self.device, layout=layout)
+        self.flip_h = ops.CoinFlip(probability = 0.5)
+        self.flip_v = ops.CoinFlip(probability = 0.5)
+        self.flip_d = ops.CoinFlip(probability = 0.5) if len(layout) == 3 else None
+
+    def define_graph(self):
+        inputs = fn.external_source(lambda: next(self.iterator))
+        inputs = 0.5 + inputs  # Make it fit the range [0.0, 1.0]
+        out = inputs.gpu() if self.device == 'gpu' else inputs
+        h = self.flip_h()
+        v = self.flip_v()
+        d = self.flip_d() if self.flip_d is not None else None
+        out = self.coord_flip(out, horizontal=h, vertical=v, depthwise=d)
+        outputs = [inputs, out, h, v]
+        if d is not None:
+            outputs.append(d)
+        return outputs
+
+def check_operator_coord_flip(device, batch_size, layout, shape):
+    eii1 = RandomDataIterator(batch_size, shape=shape, dtype=np.float32)
+    pipe = CoordFlipPipeline(device, batch_size, iter(eii1), layout)
+    pipe.build()
+    for i in range(30):
+        outputs = pipe.run()
+        for sample in range(batch_size):
+            in_coords = outputs[0].at(sample)
+            out_coords = outputs[1].at(sample)
+            h = outputs[2].at(sample)
+            v = outputs[3].at(sample)
+            d = None
+            if len(layout) == 3:
+                d = outputs[4].at(sample)
+            npoints, ndim = in_coords.shape
+
+            flip_dim = [h[0], v[0]]
+            if ndim == 3:
+                flip_dim.append(d[0])
+
+            expected_out_coords = np.copy(in_coords)
+            for d in range(ndim):
+                if flip_dim[d]:
+                    expected_out_coords[:, d] = 1.0 - in_coords[:, d]
+            np.testing.assert_allclose(out_coords[:, d], expected_out_coords[:, d])
+
+def test_operator_coord_flip():
+    for device in ['cpu']:
+        for batch_size in [1, 3]:
+            for layout, shape in [("xy", (10, 2)), ("xyz", (10, 3))]:
+                yield check_operator_coord_flip, device, batch_size, layout, shape
```

Comment:
Can you also check 1 dim as it's supported.

Reply:
I believe `0-dim` also should be supported